src/terraform-resource/out/out_lifecycle_test.go

```diff
@@ -175,4 +175,43 @@ var _ = Describe("Out Lifecycle", func() {
 		Expect(deletedVersion).To(BeTemporally(">", updatedVersion))
 		Expect(deleteOutput.Version.EnvName).To(Equal(outRequest.Params.EnvName))
 	})
+
+	It("plan infrastructure", func() {
+		planOutRequest := models.OutRequest{
+			Source: models.Source{
+				Storage: storage.Model{
+					Bucket:          bucket,
+					BucketPath:      bucketPath,
+					AccessKeyID:     accessKey,
+					SecretAccessKey: secretKey,
+					RegionName:      region,
+				},
+			},
+			Params: models.OutParams{
+				PlanOnly: true,
+				EnvName: envName,
+				Terraform: models.Terraform{
+					Source: "fixtures/aws/",
+					Vars: map[string]interface{}{
+						"access_key":     accessKey,
+						"secret_key":     secretKey,
+						"bucket":         bucket,
+						"object_key":     s3ObjectPath,
+						"object_content": "terraform-is-neat",
+						"region":         region,
+					},
+				},
+			},
+		}
+
+		By("running 'out' to create the plan file")
+
+		planRunner := out.Runner{
+			SourceDir: workingDir,
+			LogWriter: GinkgoWriter,
+		}
+		_, err := planRunner.Run(planOutRequest)
+		Expect(err).ToNot(HaveOccurred())
+
```

Comment:
I've been meaning to break the test files up a bit. Could you move the plan tests into a new test file?

It would also be nice to have a full end2end integration test for plans. For example:
- generate a plan
- expect plan file to exist on S3, expect state file to not exist
- apply the plan
- expect plan file to not exist, expect state file to exist
- expect terraform template objects were successfully created

You could have another test that verifies that you can generate a plan against an environment with an existing state file as well.